### ISSUE TITLE
Rule proposal: Prevent returning/assigning mutating array methods

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#1743]
### Description

Prevent returning or assigning mutating array methods, to improve readability and also protect against accidentally mutating an array in-place when you shouldn't. `Array#sort()`, `Array#fill()`, `Array#reverse()`.

### Fail

```js
console.log(foo.sort());
```

```js
const sorted = foo.sort();
```

### Pass

```js
foo.sort();
console.log(foo);
```

```js
foo.sort();
```

I guess we should also allow the case where there's an intermediate array?

```js
const sorted = foo.map(…).sort();
```

**Comment by sindresorhus** [comment]
> Let's forbid Array#push()( covers https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1487), Array#unshift(), Array#forEach()(unicorn/no-array-for-each already covered this, maybe people still prefer this method, but should only use as an ExpressionStatement)

👍 

> , and Array#splice() too?

Hmm, a user may need the deleted elements.

**Comment by sindresorhus** [comment]
Accepted
